instrumentation/github.com/aws/aws-lambda-go/otellambda/config.go

```diff
@@ -125,3 +133,14 @@ func WithPropagator(propagator propagation.TextMapPropagator) Option {
 		c.Propagator = propagator
 	})
 }
+
+// WithRecordError configures whether the instrumentation should record an
+// error returned by the wrapped lambda on the instrumentation-provided span.
+//
+// By default, returned errors are not recorded.
+func WithRecordError(recordError bool, recordStackTrace bool) Option {
```

Comment:
Should this be two separate options?  We don't typically accept multiple distinct configuration values in options and I don't think there's any need to check whether `c.RecordError` is set when setting `c.RecordStackTrace` as the stack trace config is only ever used when the instrumentation is configured to record errors.